README.md

```diff
@@ -655,6 +655,79 @@ val d: Seq[Int] < Any =
 
 The `Choices` effect becomes exceptionally powerful when combined with other effects. This allows you not just to make decisions or explore options in isolation but also to do so in contexts that may involve factors such as asynchronicity, resource management, or even user interaction.
 
+
+### Loops: Stack-Safe Recursion
+
+`Loops` provides a solution for general stack-safe recursion in Kyo. It offers a set of methods to transform input values through repeated applications of a function until a termination condition is met, allowing for safe and efficient recursive computations without the need for explicit effect suspensions.
+
+```scala
+import kyo._
+
+// Iteratively increment an 'Int' value
+// until it reaches 5
+val a: Int < Any = 
+  Loops.transform(1)(i => 
+    if i < 5 then Loops.continue(i + 1) 
+    else Loops.done(i)
+  )
+
+// Transform with multiple input values
+val b: Int < Any =
+  Loops.transform2(1, 1)((i, j) => 
+    if i + j < 5 then Loops.continue(i + 1, j + 1) 
+    else Loops.done(i + j)
+  )
+
+// Transform with three input values
+val c: Int < Any = 
+  Loops.transform3(1, 1, 1)((i, j, k) => 
+    if i + j + k < 5 then Loops.continue(i + 1, j + 1, k + 1)
+    else Loops.done(i + j + k)
+  )
+
+// Mixing 'IOs' with 'Loops'
+val d: Int < IOs = 
+  Loops.transform(1)(i => 
+    if i < 5 then 
+      IOs(println(s"Iteration: $i")).map(_ => Loops.continue(i + 1))
```

Comment:
Can we use `Consoles` here (or in a test)? It may be valuable to highlight that effects other than `IOs` can be combined with Loops. This may be non-obvious when not referencing the code...

Though to be honest, most newcomers may not realize Consoles is a different effect type.